### DISCUSSION TITLE
Properties.type is some enum type or can be anything?

### DISCUSSION THREAD

**Discussion by xinyuluo** [komoot/photon#673] [Q&A] [answered]
Hi everyone 👋 I wonder how `type` field under `properties` in photon response is created.
```
"properties": {
  "osm_id": 501378618,
  "osm_type": "N",
  "country": "Nigeria",
  "osm_key": "place",
  "city": "Babban Mutum",
  "countrycode": "NG",
  "osm_value": "village",
  "name": "Mutum",
  "county": "Baure",
  "state": "Katsina",
  **"type": "district"**
}
```
I've seen possible values such as house, city, district, locality, etc. Is there some predefined constants or it can be anything?

Thank you!

**Comment by mtmail** [comment]
I've counted 1000 values once, so essentially endless.

**Reply by lonvia** [reply]
You probably thinking about Nominatim and that has been fixed since https://github.com/osm-search/Nominatim/pull/2681

**Reply by xinyuluo** [reply]
@lonvia Sarah, from your link it suggests these are the possible value for `type`: the 'address level' of the object (`house`, `street`, `district`, `city`, `county`, `state`, `country`, `locality`)
https://github.com/osm-search/Nominatim/pull/2681/files#diff-a2e8bdb40ff2fc7bbd1bf5bfadbd9b741546519c4b6c03518434604138366ec3R101

Is that correct?

**Comment by lonvia** [comment, marked as answer]
The possible values that Photon returns in the `type` field can be found in the [AddressType enum](https://github.com/komoot/photon/blob/59a227b87175a32b115a8c05e3427668f1fa5bbb/src/main/java/de/komoot/photon/nominatim/model/AddressType.java#L14).
